File: OrchidCore/src/main/java/com/eden/orchid/impl/themes/templateTags/AlertTag.java
package com.eden.orchid.impl.themes.templateTags;

import com.eden.orchid.api.compilers.TemplateTag;
import com.eden.orchid.api.options.annotations.Option;
import com.eden.orchid.api.options.annotations.StringDefault;
import lombok.Getter;
import lombok.Setter;

import javax.inject.Inject;

public class AlertTag extends TemplateTag {

    @Option @StringDefault("info")
    @Getter @Setter
    private String level;

    @Option @StringDefault("")
    @Getter @Setter
    private String headline;

    @Inject
    public AlertTag() {
        super("alert", true);
    }

    @Override
    public String[] parameters() {
        return new String[] {"level", "headline"};
    }
}

File: languageExtensions/OrchidWritersBlocks/src/main/kotlin/com/eden/orchid/writersBlocks/tags/AlertTag.kt
package com.eden.orchid.writersBlocks.tags

import com.eden.orchid.api.compilers.TemplateTag
import com.eden.orchid.api.options.annotations.Option
import com.eden.orchid.api.options.annotations.StringDefault
import javax.inject.Inject

class AlertTag @Inject
constructor() : TemplateTag("alert", true) {

    @Option @StringDefault("info")
    lateinit var level: String

    @Option
    @StringDefault("")
    lateinit var headline: String

    override fun parameters(): Array<String> {
        return arrayOf("level", "headline")
    }
}
